dbms/src/Formats/JSONRowOutputStream.cpp

```diff
@@ -176,7 +176,7 @@ static void writeExtremesElement(const char * title, const Block & extremes, siz
             writeCString(",\n", ostr);
 
         writeCString("\t\t\t", ostr);
-        writeJSONString(column.name, ostr);
+        writeJSONString(column.name, ostr, settings);
```

Comment:
settings are propagated not in every call to writeJSONString.

Reply:
@alexey-milovidov please have another look, should be fine now.
Also added some basic tests.